### PR TITLE
Feature/metadata index datastore 11

### PR DESCRIPTION
**Pull request by jhedev**

### CODE REVIEW

File: pkg/datastore/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		logger: log.New().WithFields(log.Fields{
+			"component": "datastore",
+			"datastore": "dynamodb",
+		}),
+	}
+}
+
+// StoreSnapshotInfo stores the given snapshot info in the datastore
+func (d *DynamoDB) StoreSnapshotInfo(info *datastore.SnapshotInfo) error {
+
```
Comment (philippfranke) [inline pkg/datastore/dynamodb/dynamodb.go]: Please check `info == nil` otherwise it might panics

File: pkg/datastore/dynamodb/dynamodb.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+type item struct {
+	Resource  string            `dynamodbav:"snapshot_resource"`
+	CreatedAt int64             `dynamodbav:"created_at"`
+	ID        string            `dynamodbav:"snap_id"`
+	Labels    map[string]string `dynamodbav:"labels"`
+}
+
+// New creates a new DynamoDB-based datastore
+func New(client *awsdynamodb.DynamoDB, table string) *DynamoDB {
+	return &DynamoDB{
```
Comment (philippfranke) [inline pkg/datastore/dynamodb/dynamodb.go]: Please check client == nil otherwise it might panics

**Review by philippfranke — CHANGES_REQUESTED** [review]
There might be some nil pointer problems

File: pkg/datastore/dynamodb/dynamodb.go (hunk 2 above)
Comment (jhedev) [reply pkg/datastore/dynamodb/dynamodb.go]: good point

File: pkg/datastore/dynamodb/dynamodb.go (hunk 1 above)
Comment (jhedev) [reply pkg/datastore/dynamodb/dynamodb.go]: good point

**Comment by jhedev** [comment]
@philippfranke fixed and rebased

File: pkg/datastore/dynamodb/dynamodb.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		TableName: aws.String(d.table),
+		KeyConditionExpression: aws.String(
+			primaryKey + " = :snapshot_resource and " + rangeKey + " >= :created_at",
+		),
+		ExpressionAttributeValues: map[string]*awsdynamodb.AttributeValue{
+			":snapshot_resource": {
+				S: aws.String(string(resource)),
+			},
+			":created_at": {
+				N: aws.String(strconv.FormatInt(time.Now().Add(-2*24*time.Hour).Unix(), 10)),
```
Comment (gq0) [inline pkg/datastore/dynamodb/dynamodb.go:102]: ``-2*24 == -48``

File: pkg/datastore/dynamodb/dynamodb_test.go [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+		},
+	}
+
+	client := awsdynamodb.New(session.New(aws.NewConfig().WithRegion(region)))
+	t.Log("creating test table")
+	testTable, err := createTestTable(client, testTablePrefix+"_latest")
+	if err != nil {
+		t.Fatalf("createTestTable: %+v", err)
+	}
+	waitForTable(client, testTable)
```
Comment (gq0) [inline pkg/datastore/dynamodb/dynamodb_test.go]: could potentially return an error.

File: pkg/datastore/dynamodb/dynamodb_test.go [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+		},
+	}
+
+	client := awsdynamodb.New(session.New(aws.NewConfig().WithRegion(region)))
+	t.Log("creating test table")
+	testTable, err := createTestTable(client, testTablePrefix+"_delete")
+	if err != nil {
+		t.Fatalf("createTestTable: %+v", err)
+	}
+	waitForTable(client, testTable)
```
Comment (gq0) [inline pkg/datastore/dynamodb/dynamodb_test.go]: could potentially return an error.

**Review by gq0 — CHANGES_REQUESTED** [review]

File: pkg/datastore/dynamodb/dynamodb.go (hunk 3 above)
Comment (jhedev) [reply pkg/datastore/dynamodb/dynamodb.go:102]: I wanted to indicate that this is as long as two days

Comment (gq0) [reply pkg/datastore/dynamodb/dynamodb.go:102]: I see

File: pkg/datastore/dynamodb/dynamodb_test.go (hunk 4 above)
Comment (jhedev) [reply pkg/datastore/dynamodb/dynamodb_test.go]: pushed a fix

File: pkg/datastore/dynamodb/dynamodb_test.go (hunk 5 above)
Comment (jhedev) [reply pkg/datastore/dynamodb/dynamodb_test.go]: pushed a fix

**Review by gq0 — APPROVED** [review]

**Review by philippfranke — APPROVED** [review]

**Pull request merged**